queries.sql

```diff
@@ -7,3 +7,31 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered = true;
 SELECT * from animals WHERE name != 'Gabumon';
 SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK TRANSACTION;
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species != 'digimon' OR species IS NULL;
+COMMIT TRANSACTION;
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK TRANSACTION;
+
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth  > '2022-01-01';
+SAVEPOINT del2022;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO del2022;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE  weight_kg < 0;
+COMMIT TRANSACTION;
+
+SELECT COUNT (*) FROM animals;
+SELECT COUNT (*) FROM animals WHERE escape_attempts != 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered as neutered_animals , COUNT(*) as number_of_escape FROM animals WHERE escape_attempts > 0 GROUP BY neutered;
+SELECT species, MIN(weight_kg) as minimum_weight , MAX(weight_kg) as maximum_weight FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) as avg_number_of_escape FROM animals WHERE date_of_birth BETWEEN '990-01-01' AND '2000-12-31' GROUP BY species;
```

Comment:
* There is an error in the query here the year should be `1990-01-01` and not `990-01-01` Kindly make the necessary changes